packages/nx/src/plugins/js/project-graph/build-dependencies/build-dependencies.ts

```diff
@@ -19,16 +18,14 @@ export function buildExplicitDependencies(
   // TODO: TargetProjectLocator is a public API, so we can't change the shape of it
   // We should eventually let it accept Record<string, ProjectConfiguration> s.t. we
   // don't have to reshape the CreateDependenciesContext here.
-  const nodes: Record<string, ProjectGraphProjectNode> = Object.fromEntries(
-    Object.entries(ctx.projects).map(([key, config]) => [
-      key,
-      {
-        name: key,
-        type: null,
-        data: config,
-      },
-    ])
-  );
+  const nodes = {};
```

Comment:
We should keep the type-safety for the creation below:
```suggestion
  const nodes: Record<string, ProjectGraphProjectNode> = {};
```